src/hooks/useSetting.ts

```diff
@@ -0,0 +1,12 @@
+import {
+  SettingsContext,
+  SettingsContextProps,
+} from "../providers/SettingContext"
+import { useContext } from "react"
+
+// ----------------------------------------------------------------------
+
+const useSettings: () => SettingsContextProps = () =>
+  useContext(SettingsContext)
```

Comment:
It seems a little redundant to have this file. Is there anything wrong with directly calling `useContext(SettingsContext (or ThemeContext))` in the component itself?